### ISSUE TITLE
[Bug]: aws_resourcegroups_group configuration block schema is too strict

### ISSUE THREAD

**Issue by airbnb-gps** [hashicorp/terraform-provider-aws#27955]
### Terraform Core Version

1.2.3

### AWS Provider Version

4.40.0

### Affected Resource(s)

* aws_resourcegroups_group

### Expected Behavior

`configuration`'s `parameters` block should be optional.

### Actual Behavior

Schema validation currently marks `parameters` as required (https://github.com/hashicorp/terraform-provider-aws/blob/1876639e4ef78ce67b0beafa4d3f1744e6ca197f/internal/service/resourcegroups/group.go#L51).

### Relevant Error/Panic Output Snippet

```shell
Error: Insufficient parameters blocks
on <redacted> line 4, in resource "aws_resourcegroups_group" "this":
  configuration {
At least 1 "parameters" blocks are required.
```

### Terraform Configuration Files

```
resource "aws_resourcegroups_group" "this" {
  name = "name"
  configuration {
    type = "AWS::EC2::CapacityReservationPool"
  }
}
```

### Steps to Reproduce

Create an `aws_resourcegroups_group` resource having a `configuration {}` with just `type`.

### References

https://docs.aws.amazon.com/ARG/latest/APIReference/API_GroupConfigurationItem.html

`Parameters` is clearly documented as not required.

The Terraform docs also say `parameters` is optional. So the docs are correct.

The buggy code was introduced in #26934, which introduced the `configuration` block. CC @IainSteers.

**Comment by airbnb-gps** [comment]
https://docs.aws.amazon.com/ARG/latest/APIReference/about-slg.html#about-slg-types-ec2-capacityreservation shows an example involving Capacity Reservations where `Parameters` is empty.

**Comment by airbnb-gps** [comment]
This issue was at least partially addressed by #27985. I'll let @jar-b close if there's no more planned work here.

**Comment by jar-b** [comment]
I believe #27985 fixes the configuration documented here. I'll close this out, but please open a new issue (with an updated configuration) if you experience ongoing errors.
